### ISSUE TITLE
Intended behaviour?

### ISSUE THREAD

**Issue by clienthax** [zml2008/configurate#42]
The following code produces this config file when ran without a existing config file,
as im defining a default value for the boolean in the getBoolean method,
shouldn't wildCanFlee be set to this default value when saved?
```
# General settings for the world
general {
    # Wild Pokemon can try and flee from battles
    wildCanFlee=null
}
```

```
		HoconConfigurationLoader configurationLoader = HoconConfigurationLoader.builder().setFile(new File("newpixelmonconfig.hocon")).build();
		CommentedConfigurationNode mainNode = configurationLoader.load();

		CommentedConfigurationNode generalNode = mainNode.getNode("general");
		generalNode.setComment("General settings for the world");
		allowWildPokemonFleeing = generalNode.getNode("wildCanFlee").setComment("Wild Pokemon can try and flee from battles").getBoolean(false);

		configurationLoader.save(mainNode);
```

Also is there a way to make configurate not sort the map before writing?, alphabetical order kinda ruins internal ordering

**Comment by zml2008** [comment]
alphabetical sorting is impossible to fix with the hocon lib being how it is.

however, the defaults thing is something that can be fixed, probably with a ConfigurationOptions field

**Comment by zml2008** [comment]
The defaults issue has been fixed in 760cf6c

**Comment by randombyte-developer** [comment]
https://github.com/lightbend/config/pull/557 looks promising with changing the key sort order
